student.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'person'
+require_relative 'classroom'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(name = 'Unknown', age = 0, parent_permission: true, classroom: nil)
+    super(name, age, parent_permission: parent_permission)
+    self.classroom = classroom
+  end
+
+  def classroom=(classroom)
+    return if @classroom == classroom # Return early if already assigned to the same classroom
+
+    # Remove student from the current classroom
+    @classroom&.remove_student(self)
+
+    @classroom = classroom
+
+    return unless classroom
+
+    classroom.add_student(self) # Add student to the new classroom
+  end
+
+  def to_h
+    super.merge({
+                  classroom: @classroom.label # Cambiado para evitar la recursividad
+                })
+  end
```

Comment:
On exit of the app, a `noMethodError` is raised, which prevents the respective data for the `people.json` and `rentals.json` to be saved. Kindly note that your app ought to work properly and all data has to be saved in the `json` file as this is a requirement😉.

<img width="736" alt="image" src="https://github.com/maximoortelli/OOP_SCHOOL_LIBRARY_PART/assets/64643995/3be765b2-bf32-4c1d-8d5f-082deaa28386">


Please kindly ensure all errors are fixed when working with the app as this can lead to a bad user experience.

Reply:
Done, thank you @Ibizugbe for you review. You are the best💪😃